src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts

```diff
@@ -95,14 +123,35 @@ class ModalFiltersStore {
     this.currentPage = 0
     this.isAllFiltersChecked = false
 
-    this.originGroupList = toJS(dtreeStore.selectedGroups[2]) ?? []
+    this.resetCurrentMode()
+    this.currentGroupSubKind = undefined
   }
 
-  get filteredGroupList(): [string, number][] {
-    const originGroupList: [string, number][] =
-      toJS(dtreeStore.selectedGroups[2]) ?? []
+  get originGroupList(): [string, number][] {
+    const currentStepIndex = activeStepStore.activeStepIndex
+    const currentGroupIndex = dtreeModalStore.groupIndexToChange
+
+    const currentGroup =
+      dtreeStore.stepData[currentStepIndex].groups[currentGroupIndex]
+
+    if (currentGroup) {
+      const subGroups = Object.values(dtreeStore.getQueryBuilder)
+      const { groupName } = modalEditStore
+
+      const selectedGroup: StatList | undefined = subGroups
+        .find(subGroup => subGroup.find(element => element.name === groupName))
+        ?.find(element => element.name === groupName)
+
+      this.currentGroupSubKind = selectedGroup?.['sub-kind'] as SubKinds
```

Comment:
seems that you can have exception on trying cast undefined(from safe calling) to `SubKinds`

Reply:
thanks